cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -808,6 +861,119 @@ protected Future<Void> updateCertificateSecrets(List<Secret> secrets) {
+                                kafka.generateSecret(secretData, KafkaResources.internalAuthzTrustedCertsSecretName(kafka.getCluster()))))
+                .compose(ignore -> {
+                    authorizerServerCertificateHash = certHashes.stream().mapToInt(e -> e).sum();
+                    return Future.succeededFuture();
+                });
+    }
+
+    /**
+     * Copies secrets for trusted certificates used by OAuth into a single internal secret.
+     * This is used when configuring truststore for the OAuth server.
+     *
+     * @return  Future that completes when the internal Secret was successfully created or updated
+     *          or when OAuth is not used or no trusted certificates specified
+     */
+    protected Future<Void> oauthTrustedCertsSecret() {
+        Set<String> secretsToCopy = new HashSet<>();
+        if (!kafka.getListeners().isEmpty()) {
+            for (GenericKafkaListener listener : kafka.getListeners()) {
+                if (listener.getAuth() instanceof KafkaListenerAuthenticationOAuth oauth && oauth.getTlsTrustedCertificates() != null) {
+                    secretsToCopy.addAll(oauth.getTlsTrustedCertificates().stream().map(CertSecretSource::getSecretName).toList());
+                }
+            }
+        }
+
+        if (secretsToCopy.isEmpty()) {
+            return Future.succeededFuture();
+        }
+
+        List<String> certs = new ArrayList<>();
+        String oauthSecret = KafkaResources.internalOauthTrustedCertsSecretName(kafka.getCluster());
+        return Future.join(secretsToCopy.stream()
+                        .map(secretName -> secretOperator.getAsync(reconciliation.namespace(), secretName)
+                                .compose(secret -> {
+                                    if (secret == null) {
+                                        return Future.failedFuture("Secret " + secretName + " not found");
+                                    } else {
+                                        secret.getData().entrySet().stream()
+                                                .filter(e -> e.getKey().contains(".crt"))
+                                                // certificates appended under a same key so that we create a single volume mounted file
+                                                .forEach(e -> certs.add(e.getValue()));
+                                    }
+                                    return Future.succeededFuture();
+                                }))
+                        .collect(Collectors.toList()))
+                .compose(ignore -> secretOperator.reconcile(
+                                reconciliation,
+                                reconciliation.namespace(),
+                                oauthSecret,
+                                kafka.generateSecret(Map.of(oauthSecret + ".crt", mergeAndEncodeCerts(certs)), oauthSecret))
+                        .mapEmpty());
```

Comment:
the core part of this method is exactly the same as the `authzTrustedCertsSecret`. Is there any way to factor out a common method here? wdyt?